mono/mini/mini-generic-sharing.c

```diff
@@ -960,7 +966,9 @@ class_type_info (MonoDomain *domain, MonoClass *klass, MonoRgctxInfoType info_ty
 		else
 			method = mono_class_get_method_from_name (klass, "Unbox", 1);
 
-		addr = mono_compile_method (method);
+		addr = mono_jit_compile_method (method, error);
+		if (!mono_error_ok (error))
```

Comment:
Should call `mono_error_cleanup (error)` before returning, otherwise it will leak.